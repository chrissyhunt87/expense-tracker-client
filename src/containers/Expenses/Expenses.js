import React, { Component } from 'react';
import { HeaderDiv } from '../../elements/Div';

class Expenses extends Component {
    constructor() {
        super();
        this.state = {
            withinBudget: false
        }
    }

    render() {
        return (
            <React.Fragment>
                <HeaderDiv withinBudget={this.state.withinBudget}>
                    <h1>Hello, Chrissy!</h1>
                </HeaderDiv>
                {this.state.withinBudget && <p>Congratulations! You've only spent $50 this month.</p>}
                {!this.state.withinBudget && <p>Whoops! You're over budget.</p>}
            </React.Fragment>
        )
    }
}

export default Expenses;