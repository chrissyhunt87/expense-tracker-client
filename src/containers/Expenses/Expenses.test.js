import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, render } from 'enzyme';
import Expenses from './Expenses';

describe('Expenses component', () => {
    it('Component renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Expenses />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Expenses debug />);
        expect(component).toMatchSnapshot();
    })

    it('should show correct header text', () => {
        const wrapper = shallow(<Expenses />);
        expect(wrapper.find('h1').text("Hello, Chrissy!"));
    })
});