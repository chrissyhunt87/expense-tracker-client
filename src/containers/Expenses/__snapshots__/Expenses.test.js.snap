// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses component should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Expenses
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          withinBudget={false}
        >
          <h1>
            Hello, Chrissy!
          </h1>
        </ForwardRef>,
        false,
        <p>
          Whoops! You're over budget.
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1>
            Hello, Chrissy!
          </h1>,
          "withinBudget": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello, Chrissy!",
          },
          "ref": null,
          "rendered": "Hello, Chrissy!",
          "type": "h1",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "dunfGh",
            "rules": Array [
              "
    width: 100%;
    height: 15%;
    max-height: 50px;
    background-color: ",
              [Function],
              ";
    color: #FFF;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Whoops! You're over budget.",
        },
        "ref": null,
        "rendered": "Whoops! You're over budget.",
        "type": "p",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            withinBudget={false}
          >
            <h1>
              Hello, Chrissy!
            </h1>
          </ForwardRef>,
          false,
          <p>
            Whoops! You're over budget.
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1>
              Hello, Chrissy!
            </h1>,
            "withinBudget": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello, Chrissy!",
            },
            "ref": null,
            "rendered": "Hello, Chrissy!",
            "type": "h1",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "lastClassName": "dunfGh",
              "rules": Array [
                "
    width: 100%;
    height: 15%;
    max-height: 50px;
    background-color: ",
                [Function],
                ";
    color: #FFF;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Whoops! You're over budget.",
          },
          "ref": null,
          "rendered": "Whoops! You're over budget.",
          "type": "p",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
